src/main/java/qna/service/QnAService.java

```diff
@@ -38,21 +35,9 @@ public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteE
         if (!question.isOwner(loginUser)) {
             throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
         }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
+        if(!question.isAllAnswersOwner(loginUser)){
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
         }
-        deleteHistoryService.saveAll(deleteHistories);
+        deleteHistoryService.saveAll(question.delete().get());
```

Comment:
`question.delete().get()`만 봐서는 무엇을 반환하는지 알기 쉽지 않네요. 
로직을 작성할 때 가장 좋은건 다른 객체나 계층을 트레이싱하지 않고 해당 로직만 보고도 이게 무엇을 의미하는지 파악할 수 있는게 베스트입니다. 
```suggestion
        DeleteHistories deleteHistories = question.delete();
        deleteHistoryService.saveAll(deleteHistories.getAll());
```
이런 식이 좀 더 가독성이 좋지 않을까 싶어요 ㅎㅎ